src/sc/sidechain.cpp

```diff
@@ -757,61 +778,139 @@ void CSidechain::UpdateScFees(const CScCertificateView& certView)
     LogPrint("sc", "%s():%d - pushing f=%d/m=%d into list with size %d\n",
         __func__, __LINE__, ftScFee, mbtrScFee, scFees.size());
 
-    scFees.push_back(Sidechain::ScFeeData(ftScFee, mbtrScFee));
+    // v1 sidechains
+    if (!isNonCeasing()) {
+        scFees.push_back(Sidechain::ScFeeData(ftScFee, mbtrScFee));
 
-    // check if we are past the buffer size
-    int delta = scFees.size() - maxSizeOfScFeesContainers;
-    if (delta > 0)
-    {
-        // remove from the front as many elements are needed to be within the circular buffer size
-        // --
-        // usually this is just one element, but in regtest a node can set the max size via a startup option
-        // therefore such size might be lesser than scFee size after a node restart
-        while (delta > 0)
+        int delta = scFees.size() - maxSizeOfScFeesContainers;
```

Comment:
This is dangerous for various reasons:

- sign mismatch amongst operators
- implicit narrowing conversion from size_t to int

Raising warnings to errors in compiler causes a fail cause what the compiler actually does is `(int)(scFees.size() - ((size_t)maxSizeOfScFeesContainers))` which may cause an underflow (e.g. 1-5) which can return a "meaningful result" only in some cases.
See https://godbolt.org/z/37eTsafrf

Reply:
`delta` variable has been removed in 2522a6032b25bc34777cfd408f489058fb189cee